osu.Game/Screens/Play/BreakOverlay.cs

```diff
@@ -109,6 +121,41 @@ protected override void LoadComplete()
             initializeBreaks();
         }
 
+        protected override void Update()
+        {
+            base.Update();
+
+            updateBreakTimeBindable();
+        }
+
+        private void updateBreakTimeBindable()
+        {
+            if (breaks == null || !breaks.Any())
+            {
+                isBreakTime.Value = false;
+                return;
+            }
+
+            int indexDirection = Clock.CurrentTime < breaks[currentBreakIndex].StartTime ? -1 : (Clock.CurrentTime > breaks[currentBreakIndex].EndTime ? 1 : 0);
+
+            while (Clock.CurrentTime < breaks[currentBreakIndex].StartTime || Clock.CurrentTime > breaks[currentBreakIndex].EndTime)
+            {
+                currentBreakIndex += indexDirection;
+
+                if (currentBreakIndex < 0 || currentBreakIndex >= breaks.Count)
+                    break;
+            }
+
+            if (currentBreakIndex < 0 || currentBreakIndex >= breaks.Count)
```

Comment:
work through in your head or on paper what happens when the second condition here becomes true

Reply:
break time will be false and the index will be reset, it's a check if the index was out of range, is there something i'm missing?

Reply:
and next loop?

Reply:
oh, I understand you now, I think saving an index before the while loop and assigning it back might do it